lib/Driver/ToolChains.cpp

```diff
@@ -1452,9 +1452,11 @@ void ToolChain::getClangLibraryPath(const ArgList &Args,
   getResourceDirPath(LibPath, Args, /*Shared=*/true);
   // Remove platform name.
   llvm::sys::path::remove_filename(LibPath);
-  llvm::sys::path::append(LibPath, "clang", "lib",
-                          T.isOSDarwin() ? "darwin"
-                                         : getPlatformNameForTriple(T));
+  auto platformName =
+      T.isOSDarwin() ? "darwin" : getPlatformNameForTriple(Triple);
+  if (platformName == "android")
+    platformName = "linux";
+  llvm::sys::path::append(LibPath, "clang", "lib", platformName);
```

Comment:
This is a translation of apple/swift-driver#1372, see that pull for more info.

Reply:
I wonder if we can use `llvm::Triple::getOSName` instead.

Reply:
> I wonder if we can use llvm::Triple::getOSName instead.

Just looked into this: [clang uses `ToolChain::getOSLibName()` instead](https://github.com/llvm/llvm-project/blob/bbc0f99f3bc96f1db16f649fc21dd18e5b0918f6/clang/lib/Driver/ToolChain.cpp#L607), which [has this same darwin override](https://github.com/llvm/llvm-project/blob/bbc0f99f3bc96f1db16f649fc21dd18e5b0918f6/clang/lib/Driver/ToolChain.cpp#L577). We can't call that without instantiating a `clang::ToolChain()` though, and [given all the ceremony you added to do so elsewhere](https://github.com/apple/swift/blob/eee9331d97c348c87e8ae52901c7f0c3309d9a39/lib/ClangImporter/ClangIncludePaths.cpp#L440), that would end up with more lines.

@compnerd, I don't think that clang method is worth invoking, let me know if you disagree.